src/key.cpp

```diff
@@ -350,28 +350,90 @@ bool CKey::SetCompactSignature(uint256 hash, const std::vector<unsigned char>& v
     return false;
 }
 
-bool CKey::Verify(uint256 hash, const std::vector<unsigned char>& vchSigParam)
+static bool ParseLength(
+        const std::vector<unsigned char>::iterator& begin,
+        const std::vector<unsigned char>::iterator& end,
+        size_t& nLengthRet,
+        size_t& nLengthSizeRet)
+{
+    std::vector<unsigned char>::iterator it = begin;
+    if (it == end)
+        return false;
+
+    nLengthRet = *it;
+    nLengthSizeRet = 1;
+
+    if (!(nLengthRet & 0x80))
+        return true;
+
+    unsigned char nLengthBytes = nLengthRet & 0x7f;
+
+    nLengthRet = 0;
+    for (unsigned char i = 0; i < nLengthBytes; i++)
+    {
+        it++;
+        if (it == end)
+            return false;
+        nLengthRet = (nLengthRet << 8) | *it;
+        if (nLengthRet > 0x7f)
```

Comment:
scriptSig currently allows up to 500 bytes (c.f. CTransaction::IsStandard()), so here the logic probably should be more flexible, avoid introducing hidden constraint on scriptSig. Something like 
if (nLengthRet > 0x7fffff)
should work